Improve styling for bot editing modals

ayazhan8 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
PR for [issue#21410: Improve styling for bot editing modals ](https://github.com/zulip/zulip/issues/21410)

[The first commit: ](https://github.com/zulip/zulip/commit/f13864c3ac734c3449068ba2f17b6092486c002b)
- Adds space between "Bot Email" and "Owner" fields in "Edit bot" modal (screenshot1).
- Removes font styles for field labels. "Edit bot" and "Change bot info and owner" modals now use regular font for field labels.

[The second commit: ](https://github.com/zulip/zulip/commit/923a7c254f36181fc8581048c277bdd29fc042ed)
- Fixes the vertical spacing between input fields.

**Testing plan:**
Manually

**GIFs or screenshots:** 
<img width="380" alt="image" src="https://user-images.githubusercontent.com/54767127/158469004-d63a3446-9c83-4ee0-89fc-dd0e61839f1c.png">
Screenshot1


<img width="385" alt="image" src="https://user-images.githubusercontent.com/54767127/158467168-de66dd31-3b20-4e3c-9f49-8725ca7bfba4.png">
Screenshot2

alya commented:
The screenshots look good to me! @sahil839 would you be up for reviewing this one?

sahil839 commented:
Changes looks good to me.Also I think the explaination written in the PR comment can be copied in the commit message.

Also, in "Edit bot" we currently show email as plain text and in "Change bot info and owner" modal we show a disabled input. Do we want to make them consistent? Also probably the order of fields in both the modals, with avatar button, being the only different field, at the end.

ganpa3 commented on static/templates/settings/admin_bot_form.hbs:
```diff
@@ -7,11 +7,11 @@
             <input type="text" autocomplete="off" name="full_name" value="{{ full_name }}" />
         </div>
-        <div class="email_change_container">
+        <div class="input-group email_change_container">
```
I think you can remove this change, since the `email_change_container` already has enough space below it, and adding more makes it look too spaced out.

ayazhan8 replied:
Removed, thank you!

ganpa3 commented:
The new design looks too spaced out IMO.

ganpa3 commented:
Also, the text in "Bot email" feels cut out from below.

ayazhan8 commented:
@ganpa3 Ready for another review. Please let me know if I need to change anything.


> Also, the text in "Bot email" feels cut out from below. ![image](https://user-images.githubusercontent.com/58626718/160169232-c2396ce2-1616-4c00-bc49-29f4c26805dd.png)

I did not quite understand it. Could you please tell me if I should change something or submit this PR?

alya commented:
I think we can rename "Full name" -> "Name". @sahil839 does that sound right?

ayazhan8 commented:
@alya @sahil839 Ready for another review.
1. I squashed my commits
2. Changed the email field to be a disabled input on both forms, and
3. Reordered the fields
Here are the screenshots:
<img width="425" alt="zulip-issue1" src="https://user-images.githubusercontent.com/54767127/162502736-a1686a41-9bcf-4869-b795-a7165245d72d.png">
<img width="422" alt="zulip-issue" src="https://user-images.githubusercontent.com/54767127/162502871-65519490-079f-4982-9441-4d6686399c13.png">

ayazhan8 commented:
@alya @sahil839 Ready for another review. Is there anything left to change?

sahil839 commented on static/templates/settings/admin_bot_form.hbs:
```diff
@@ -1,19 +1,19 @@
             <input type="text" autocomplete="off" name="full_name" value="{{ full_name }}" />
         </div>
-        <div class="email_change_container">
+        <div class="input-group email_change_container">
```
I think we can rename this class, since we cannot actually change the email.

TonaDaphin replied:
Renamed it to "email_container".
Thanks

sahil839 commented on static/styles/settings.css:
```diff
@@ -960,7 +957,7 @@ input[type="checkbox"] {
     text-align: left;
     margin-top: 0;
-    margin-bottom: 10px;
+    margin-bottom: 20px;
```
This CSS is not used anywhere now since you removed the `edit_bot_email` class it is not used anywhere else, which is kinda right since we cannot change the email, so we can actually remove this CSS.

TonaDaphin replied:
Removed the property

sahil839 commented on static/templates/settings/edit_bot.hbs:
```diff
@@ -2,22 +2,22 @@
+                <div><label for="edit_bot_name" generated="true" class="text-error"></label></div>
+            </div>
+            <div class="input-group">
+                <label for="bot_email">{{t "Bot email" }}</label>
```
@alya Can we also change the heading here to be `Email` like we do in other modal?

alya replied:
Yes, definitely -- good catch!

sahil839 reviewed (commented):
Posted a few comments.

```quote
I think we can rename "Full name" -> "Name". @sahil839 does that sound right?
```
Not sure, we use `Full name` in other places like in `Manage user` form and even in form for creating bots. May be we should change it at all places (either all places for bots or in general) if we are doing so.

alya commented:
> Not sure, we use `Full name` in other places like in `Manage user` form and even in form for creating bots. May be we should change it at all places (either all places for bots or in general) if we are doing so.

Ah, right. Yes, let's change it everywhere for bots. @TonaDaphin could you please change "Full name" -> "Name" here and on the "Add a new bot" form as well?

![Screen Shot 2022-04-25 at 11 11 29 PM](https://user-images.githubusercontent.com/2090066/165233670-d85e10c3-fcec-4f88-bfea-ae946907e6d5.png)

sahil839 commented:
The issue for which this PR was opened is already closed. We discussed about changing the "Full name" label to "Name" for the bot modals and that is the only change left. I can close this PR and open a new one just with the label change from "Full name" to "Name" if we still want to do that change. @alya FYI, just want to confirm before opening the new PR.
